hushline/routes.py

```diff
@@ -29,9 +29,12 @@
 
 
 def valid_username(form: Form, field: Field) -> None:
-    if not re.match(r"^[a-zA-Z0-9_-]+$", field.data):
+    # Corrected regex pattern with length constraint (4 to 20 characters)
+    if not re.match(r"^[a-zA-Z0-9]([a-zA-Z0-9_-]{2,18}[a-zA-Z0-9])?$", field.data):
```

Comment:
I think wrapping the new part of the regex in `(...)?` means that it'll match if there are 0 or 1 instances, meaning that this will also match a single character username. This might be more like what you are looking for:

```suggestion
    if not re.match(r"^[a-zA-Z0-9][a-zA-Z0-9_-]{2,18}[a-zA-Z0-9]$", field.data):
```

I'd definitely recommend writing a test for this function.